spec.bs

```diff
@@ -1505,6 +1513,46 @@ partial interface Navigator {
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Navigator>deprecatedURNtoURL(|urnOrConfig|, |send_reports|)</dfn>
+  method steps are:
+
+  1. Let |urn| be null.
+
+  1. If |urnOrConfig| is a {{USVString}}, set |urn| to |urnOrConfig|.
+
+  1. Otherwise, set |urn| to |urnOrConfig|'s [=fencedframeconfig/urn=].
+
+  1. If |urn| is <span class=XXX>TODO invalid</span>, [=exception/throw=] a {{TypeError}}.
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |global| be [=this=]'s [=relevant global object=].
+
+  1. Run the following steps [=in parallel=]:
+
+     1. Let |mapping| be |global|'s [=associated Document=]'s [=node navigable=]'s
+        [=navigable/traversable navigable=]'s [=traversable navigable/fenced frame config mapping=].
+
+     1. Let |config| be the result of [=fenced frame config mapping/finding a config=] in |mapping|
```

Comment:
Does that mean callers have to wait for a Protected Audience auction to complete before calling `deprecatedURNtoURL()`?
It does look like we require the mapping to be finalized: https://source.chromium.org/chromium/chromium/src/+/main:content/browser/interest_group/ad_auction_service_impl.cc;l=487;drc=2b8b8c15ea96feadd34e04ca947e5f05069e3b24
Should I just change this to instead say this?
```
    1. If |mapping|'s [=fenced frame config mapping/finalized config mapping=][|urn|], does not
       [=map/exist=], [=queue a global task=] on the [=DOM manipulation task source=]
       given |global|, to [=resolve=] |p| with {{undefined}}, and abort these steps.
    1. Let |config| be |mapping|'s [=fenced frame config mapping/finalized config mapping=][|urn|].
```

Reply:
It's not an issue for PA because the pending urn is never returned to script, so it can't be passed to deprecatedURNtoTURL; instead the promise waits for the auction result to be resolved. In selectURL it immediately resolves to a pending urn, so you could pass it as input (and it won't work).

Reply:
So what's the right thing to put in the spec?  it sounds like it only works with finalized URNs, in which case I think my suggestion in my last comment is accurate?

Reply:
Yeah, afaict what you put in the comment matches the implementation.

Reply:
Okay, I switched to my suggested steps.